src/main/java/com/atlassian/jira/cloud/jenkins/deploymentinfo/service/ChangeLogIssueKeyExtractor.java

```diff
@@ -18,11 +26,21 @@
  */
 public final class ChangeLogIssueKeyExtractor implements IssueKeyExtractor {
 
+    private static final Logger log = LoggerFactory.getLogger(ChangeLogIssueKeyExtractor.class);
+
     public Set<String> extractIssueKeys(final WorkflowRun workflowRun) {
 
         final Set<IssueKey> allIssueKeys = new HashSet<>();
         final List<ChangeLogSet<? extends ChangeLogSet.Entry>> changeSets =
-                workflowRun.getChangeSets();
+                new ArrayList<>(workflowRun.getChangeSets());
+
+        WorkflowRun previous = workflowRun.getPreviousBuild();
+        while (Objects.nonNull(previous) && !isBuildSuccessful(previous)) {
```

Comment:
@srnjak, May I know why are we checking for !isBuildSuccessful(previous)? when I tested the code, ChangeLogIssueKeyExtractor is not picking issue keys from last successful build.
Scenario I tested is as follows:
[Build 1]:

- Committed a change from gitHub branch [Test-123] and added an issue key [Test-456] in commit title.

- build succeeded and sent build info to both branch Jira issue [Test-123] and commit title [Test-456] Jira issue.

[Build 2]:

- Build the current branch [Test-123] and test succeeded.
- [Expectation]: Both Jira Issues [Test-123] and [Test-456] should  update with build Info.
- [Actual Result]: Only Issue [Test-123] updated with build Info. 

Let me know why it is not working or I am testing it wrong.

Reply:
We want to pick the current build and all previous builds until we find the successful one.

**Scenario 1**
* #86 SUCCESS (current build)
* #85 SUCCESS
* #84 SUCCESS

We pick the keys from the build #86 only.

**Scenario 2**
* #86 SUCCESS (current build)
* #85 FAILED
* #84 SUCCESS

We pick the keys from the builds #86 and #85 as well. The build #84 is not interesting for us since all the issues had been resolved until then.

The key from the branch will always be picked because the whole branch is about the issue.

Reply:
@srnjak, Thanks for the explanation. It makes sense. We will do a testing blitz by tomorrow, if everything  goes fine. We will be able to release latest by Monday.